res/controllers/midi-components-0.0.js

```diff
@@ -715,6 +715,67 @@
         }
     });
 
+    var JogWheel = function(options) {
```

Comment:
because this class does not properly set the scratch2 parameter, can you call it "JogWheelBasic" instead?  And add documentation to say that it will not behave at maximum performance.  Then we can later create a new JogWheelPrecise (or something like that) class that properly uses timers and scratch2.   (I can try doing that be refactoring the code I wrote for the traktor S3, which has been duplicated elsewhere)

Reply:
The issue with our scratching code right now is IMO thats its not documented very well. If you'd explain the better API to me, I'd prefer to implement it here right away.

Reply:
Not all jog wheels work the same way.  I think this PR is valuable, and we'll need more than one API to map the different types of wheels. Not all controllers will be able to use the more complex APIs.  Let's just call this one JogWheelBasic to differentiate it from other APIs.  

The next jogwheel API would be called "JogWheelClocked" or something like that, to describe that the controller HID message itself has a clock value to calculate jog wheel timing (rather than using internal mixxx timings).  The NI controllers have this, and others might too but I don't think we've checked.  See here:

https://github.com/mixxxdj/mixxx/blob/main/res/controllers/Traktor-Kontrol-S3-hid-scripts.js#L628 and below, esp the wheeldeltas function.  Magic numbers abound, and some of them can be constants but some of them have to be passed in.  I would recommend an object-based jog controller that takes parameters to a constructor and then has accessor functions to give it input and calculate output.